bug: Appium 2 fails to install drivers

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When I try to install a driver (any driver), it fails with the following:

```
$ appium driver install xcuitest
✖ Installing 'xcuitest' using NPM install spec 'appium-xcuitest-driver'
Error: ✖ Encountered an error when installing package: npm command 'install --save-dev --omit=peer --save-exact --global-style --no-package-lock appium-xcuitest-driver --json' failed with code null.

STDOUT:


STDERR:
```

### Expected Behavior

It works, or at least provides me with some output.

### Minimal Reproducible Example

N/A

### Environment

- Operating system: macos
- If running via `appium` CLI...
  - Appium CLI version (output of `appium --version`): 2.0.0-beta.48
  - Node.js version (output of `node --version`): 16.16
  - `npm` version (output of `npm --version`):  8.11 / Yarn 3
  - Last `appium` version which did _not_ exhibit the problem: N/A
- If running Appium Desktop, its version:
- Appium driver(s) and their version(s):
- Appium plugin(s) and their version(s):
- Platform and version under test:
- Real device or emulator/simulator: simulator
 

### Link to Appium Logs

_No response_

### Futher Information

_No response_

Ok, wiping `~/.appium` seemed to do the trick. Was able to install xcuitest and uiautomator2, however, chromium fails.

```
✖ Installing 'chromium' using NPM install spec 'appium-chromium-driver'
Error: ✖ Encountered an error when installing package: npm command 'install --save-dev --omit=peer --save-exact --global-style --no-package-lock appium-chromium-driver --json' failed with code 1.

STDOUT:


STDERR:
npm ERR! code 1
npm ERR! path /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver
npm ERR! command failed
npm ERR! command sh -c node install-npm.js
npm ERR! [15:32:16] Error installing Chromedriver: Cannot find module '@appium/types'
npm ERR! Require stack:
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/build/lib/basedriver/driver.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/build/lib/index.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/index.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/utils.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/storage-client.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/install.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/install-npm.js
npm ERR! [15:32:16] Error: Cannot find module '@appium/types'
npm ERR! Require stack:
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/build/lib/basedriver/driver.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/build/lib/index.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/index.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/utils.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/storage-client.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/install.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/install-npm.js
npm ERR!     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
npm ERR!     at Function.Module._load (node:internal/modules/cjs/loader:778:27)
npm ERR!     at Module.require (node:internal/modules/cjs/loader:1005:19)
npm ERR!     at require (node:internal/modules/cjs/helpers:102:18)
npm ERR!     at Object.<anonymous> (/Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/lib/basedriver/driver.js:18:1)
npm ERR!     at Module._compile (node:internal/modules/cjs/loader:1105:14)
npm ERR!     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
npm ERR!     at Module.load (node:internal/modules/cjs/loader:981:32)
npm ERR!     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
npm ERR!     at Module.require (node:internal/modules/cjs/loader:1005:19)
npm ERR!     at require (node:internal/modules/cjs/helpers:102:18)
npm ERR!     at Object.<anonymous> (/Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/lib/index.js:8:1)
npm ERR!     at Module._compile (node:internal/modules/cjs/loader:1105:14)
npm ERR!     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
npm ERR!     at Module.load (node:internal/modules/cjs/loader:981:32)
npm ERR!     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
npm ERR! [15:32:16] Downloading Chromedriver can be skipped by setting the'APPIUM_SKIP_CHROMEDRIVER_INSTALL' environment variable.
{
  "error": {
    "code": 1,
    "summary": "command failed",
    "detail": "sh -c node install-npm.js\n[15:32:16] Error installing Chromedriver: Cannot find module '@appium/types'\nRequire stack:\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/build/lib/basedriver/driver.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/build/lib/index.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/index.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/utils.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/storage-client.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/install.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/install-npm.js\n[15:32:16] Error: Cannot find module '@appium/types'\nRequire stack:\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/build/lib/basedriver/driver.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/build/lib/index.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/index.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/utils.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/storage-client.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/install.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/install-npm.js\n    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)\n    at Function.Module._load (node:internal/modules/cjs/loader:778:27)\n    at Module.require (node:internal/modules/cjs/loader:1005:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at Object.<anonymous> (/Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/lib/basedriver/driver.js:18:1)\n    at Module._compile (node:internal/modules/cjs/loader:1105:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)\n    at Module.load (node:internal/modules/cjs/loader:981:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:822:12)\n    at Module.require (node:internal/modules/cjs/loader:1005:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at Object.<anonymous> (/Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/lib/index.js:8:1)\n    at Module._compile (node:internal/modules/cjs/loader:1105:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)\n    at Module.load (node:internal/modules/cjs/loader:981:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:822:12)\n[15:32:16] Downloading Chromedriver can be skipped by setting the'APPIUM_SKIP_CHROMEDRIVER_INSTALL' environment variable."
  }
}

```

This is on 8.11:

```
 npm uninstall -g appium && npm install -g appium@next
rm -rf ~/.appium
appium driver install xcuitest
appium driver install uiautomator2
appium driver install chromium
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.

removed 453 packages, and audited 1 package in 1s

found 0 vulnerabilities
npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead.

added 453 packages, and audited 454 packages in 12s

49 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
✔ Installing 'xcuitest' using NPM install spec 'appium-xcuitest-driver'
ℹ Driver xcuitest@4.16.6 successfully installed
- automationName: XCUITest
- platformNames: ["iOS","tvOS"]
✔ Installing 'uiautomator2' using NPM install spec 'appium-uiautomator2-driver'
ℹ Driver uiautomator2@2.12.1 successfully installed
- automationName: UiAutomator2
- platformNames: ["Android"]
✖ Installing 'chromium' using NPM install spec 'appium-chromium-driver'
Error: ✖ Encountered an error when installing package: npm command 'install --save-dev --omit=peer --save-exact --global-style --no-package-lock appium-chromium-driver --json' failed with code 1.

STDOUT:


STDERR:
npm ERR! code 1
npm ERR! path /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver
npm ERR! command failed
npm ERR! command sh -c node install-npm.js
npm ERR! [11:54:36] Error installing Chromedriver: Cannot find module '@appium/types'
npm ERR! Require stack:
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/build/lib/basedriver/driver.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/build/lib/index.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/index.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/utils.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/storage-client.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/install.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/install-npm.js
npm ERR! [11:54:36] Error: Cannot find module '@appium/types'
npm ERR! Require stack:
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/build/lib/basedriver/driver.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/build/lib/index.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/index.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/utils.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/storage-client.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/install.js
npm ERR! - /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/install-npm.js
npm ERR!     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
npm ERR!     at Function.Module._load (node:internal/modules/cjs/loader:778:27)
npm ERR!     at Module.require (node:internal/modules/cjs/loader:1005:19)
npm ERR!     at require (node:internal/modules/cjs/helpers:102:18)
npm ERR!     at Object.<anonymous> (/Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/lib/basedriver/driver.js:18:1)
npm ERR!     at Module._compile (node:internal/modules/cjs/loader:1105:14)
npm ERR!     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
npm ERR!     at Module.load (node:internal/modules/cjs/loader:981:32)
npm ERR!     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
npm ERR!     at Module.require (node:internal/modules/cjs/loader:1005:19)
npm ERR!     at require (node:internal/modules/cjs/helpers:102:18)
npm ERR!     at Object.<anonymous> (/Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/lib/index.js:8:1)
npm ERR!     at Module._compile (node:internal/modules/cjs/loader:1105:14)
npm ERR!     at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
npm ERR!     at Module.load (node:internal/modules/cjs/loader:981:32)
npm ERR!     at Function.Module._load (node:internal/modules/cjs/loader:822:12)
npm ERR! [11:54:36] Downloading Chromedriver can be skipped by setting the'APPIUM_SKIP_CHROMEDRIVER_INSTALL' environment variable.
{
  "error": {
    "code": 1,
    "summary": "command failed",
    "detail": "sh -c node install-npm.js\n[11:54:36] Error installing Chromedriver: Cannot find module '@appium/types'\nRequire stack:\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/build/lib/basedriver/driver.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/build/lib/index.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/index.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/utils.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/storage-client.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/install.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/install-npm.js\n[11:54:36] Error: Cannot find module '@appium/types'\nRequire stack:\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/build/lib/basedriver/driver.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/build/lib/index.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/index.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/utils.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/storage-client.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/build/lib/install.js\n- /Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/appium-chromedriver/install-npm.js\n    at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)\n    at Function.Module._load (node:internal/modules/cjs/loader:778:27)\n    at Module.require (node:internal/modules/cjs/loader:1005:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at Object.<anonymous> (/Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/lib/basedriver/driver.js:18:1)\n    at Module._compile (node:internal/modules/cjs/loader:1105:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)\n    at Module.load (node:internal/modules/cjs/loader:981:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:822:12)\n    at Module.require (node:internal/modules/cjs/loader:1005:19)\n    at require (node:internal/modules/cjs/helpers:102:18)\n    at Object.<anonymous> (/Users/milesjohnson/.appium/node_modules/appium-chromium-driver/node_modules/@appium/base-driver/lib/index.js:8:1)\n    at Module._compile (node:internal/modules/cjs/loader:1105:14)\n    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)\n    at Module.load (node:internal/modules/cjs/loader:981:32)\n    at Function.Module._load (node:internal/modules/cjs/loader:822:12)\n[11:54:36] Downloading Chromedriver can be skipped by setting the'APPIUM_SKIP_CHROMEDRIVER_INSTALL' environment variable."
  }
}

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/milesjohnson/.npm/_logs/2023-01-05T19_54_33_028Z-debug-0.log
```

--- a/packages/base-driver/package.json
+++ b/packages/base-driver/package.json
@@ -43,6 +43,7 @@
   },
   "dependencies": {
     "@appium/support": "^3.0.1",
+    "@appium/types": "^0.7.0",
     "@colors/colors": "1.5.0",
     "@types/async-lock": "1.3.0",
     "@types/bluebird": "3.5.38",

